pkg/apis/core/types.go

```diff
@@ -2190,7 +2190,8 @@ type ResourceRequirements struct {
 	// This is an alpha field and requires enabling the
 	// DynamicResourceAllocation feature gate.
 	//
-	// This field is immutable.
+	// This field is immutable. It can only be set for containers.
+	// It gets ignored in PersistentVolumeClaimSpec.
```

Comment:
This type is used from ~hundreds of out of tree APIs as well ... https://grep.app/search?q=.ResourceRequirements%20%60json%3A%22&filter[lang][0]=Go&filter[path.pattern][0]=types.go and adding a pod-specific field to it makes ~all of those incorrect / at risk for persisting nonsense / unbounded / unvalidated data.

I'm not sure about this doc change specific to PVC. Clearing the field to nil from PVC / ephemeral volumes is probably all I would do for this PR and the 1.26 backport.

Reply:
This is the documentation for the Kubernetes core API. IMHO describing how that field is used in that API makes sense. But I don't feel strongly about this and can take it out... done.

That out-of-tree APIs use these structs raises interesting questions. If they do that because the entire field is needed for the Kubernetes v1 API, then it makes sense. But if they reuse them for something else, like PVC did, then it's problematic because it implies that we cannot extend "our" struct without breaking out-of-tree code.

> I'm wondering if we should switch to isolate the ResourceRequirements type used by pods and only keep the claim field in that type in 1.27

I think we should try. Let's merge this PR and I'll get to that next.